### ISSUE TITLE
[MSVC] opencv failed to run test Imgproc_EMD on x86 mode

### ISSUE THREAD

**Issue by Zhaojun-Liu** [opencv/opencv#25601]
### System Information

OpenCV commit we used: a4486c0
OS: Windows server 2022
Compiler: msvc latest

### Detailed description

OpenCV failed to run test, and failed tests as below:
```
[  FAILED  ] 5 tests, listed below:
[  FAILED  ] Imgproc_EMD.regression
[  FAILED  ] Imgproc_EMD.distance_types
[  FAILED  ] Imgproc_EMD_dist.random_flow_verify/0, where GetParam() = 1
[  FAILED  ] Imgproc_EMD_dist.random_flow_verify/1, where GetParam() = 2
[  FAILED  ] Imgproc_EMD_dist.random_flow_verify/2, where GetParam() = 3
```

Note:
This issue only occurred on x86 mode, amd64 is ok. And I found that this issue occurred from commit 7e56908306934d6e1c337e27a8473d6ea86a8623, the previous commit 5b0843728ebd96196dbeef31cfa81c37f04234d5 is ok.

### Steps to reproduce

1. git -C "C:\gitP" clone --no-checkout https://github.com/opencv/opencv opencv\opencv
2. git -C "C:\gitP\opencv\opencv" checkout a4486c0
3. git -C "C:\gitP\opencv\opencv" submodule sync
4. git -C "C:\gitP\opencv\opencv" submodule foreach git clean -xdf
5. git -C "C:\gitP\opencv\opencv" submodule update --init --recursive
6. git -C "C:\gitP" clone --no-checkout https://github.com/opencv/opencv_extra opencv\opencv_extra
7.  git -C "C:\gitP\opencv\opencv_extra" checkout 1458fff
8. mkdir C:\gitP\opencv\opencv\build_x86 & cd /d C:\gitP\opencv\opencv\build_x86
9. cmake -G "Visual Studio 17 2022" -A Win32 -DCMAKE_SYSTEM_VERSION=10.0.22621.0 -DWITH_IPP=ON -DBUILD_SHARED_LIBS=OFF -DBUILD_PERF_TESTS=ON -DBUILD_TESTS=ON -DBUILD_EXAMPLES=ON -DWITH_OPENCL=OFF -DBUILD_DOCS=OFF -DWITH_CUDA=OFF -DWITH_EIGEN=OFF ..
10. msbuild /m /p:Platform=Win32 /p:Configuration=Release OpenCV.sln /t:Rebuild /p:BuildInParallel=true
11. set OPENCV_TEST_DATA_PATH=C:\gitP\opencv\opencv_extra\testdata
12. cd /d C:\gitP\opencv\opencv\build_x86\bin\Release
13. opencv_test_imgproc.exe --gtest_filter=-HoughCirclesTest.DefaultMaxRadius:HoughCirclesTest.CentersOnly:HoughCirclesTest.ManySmallCircles:ImgProc/HoughCirclesTestFixture.regression/0:Imgproc_GrabCut.regression

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [ ] I updated to the latest OpenCV version and the issue is still there
- [ ] There is reproducer code and related data files (videos, images, onnx, etc)

**Comment by asmorkalov** [comment]
@Zhaojun-Liu Could you try current 4.x branch. Most probably fixed in https://github.com/opencv/opencv/pull/25583

**Comment by Zhaojun-Liu** [comment]
> @Zhaojun-Liu Could you try current 4.x branch. Most probably fixed in #25583

Ok, thanks, I will try it. Once the results are in, I will update here.

**Comment by Zhaojun-Liu** [comment]
> @Zhaojun-Liu Could you try current 4.x branch. Most probably fixed in #25583

Hi @asmorkalov, I tried it, it passed on both amd64 and x86. Thank you.
